internal/validator/cosignvalidator/cosign_validator.go

```diff
@@ -106,16 +115,20 @@ func (cv *CosignValidator) UnmarshalYAML(unmarshal func(interface{}) error) erro
 		return fmt.Errorf("no trust roots provided for validator %s", valData.Name)
 	}
 
-	if valData.Cert != "" {
-		certPool := x509.NewCertPool()
-		if !certPool.AppendCertsFromPEM([]byte(valData.Cert)) {
-			return fmt.Errorf("invalid certificate for validator %s", valData.Name)
-		}
+	if valData.Cert != "" && utils.ValidateCertificate(valData.Cert) != nil {
+		return fmt.Errorf("invalid certificate for validator %s", valData.Name)
+	}
+
+	if valData.Host.FulcioCert != "" && utils.ValidateCertificate(valData.Host.FulcioCert) != nil {
+		return fmt.Errorf("invalid fulcio certificate for validator %s", valData.Name)
 	}
 
 	cv.Name = valData.Name
 	cv.Type = valData.Type
 	cv.Rekor = valData.Host.Rekor
+	cv.RekorPubkey = valData.Host.RekorPubkey
+	cv.FulcioCert = valData.Host.FulcioCert
+	cv.CTLogPubkey = valData.Host.CTLogPubkey
```

Comment:
Is there no way to validate the pub keys?

Reply:
added tests for rekor pub key and ctlog pubkey.